src/_tutorials/dart-vm/httpserver.md

```diff
@@ -780,35 +776,38 @@ It responds to all requests by returning the contents of the
 
 Here's the code for mini file server:
 
+<?code-excerpt "httpserver/bin/mini_file_server.dart"?>
 {% prettify dart %}
+import 'dart:async';
 import 'dart:io';
 
-main() async {
+File targetFile = new File('index.html');
+
+Future main() async {
```

Comment:
I kind of hate seeing `main` annotated with a type. It obscures the entry point for little to no benefit, as far as I can see.

Does the analyzer complain if we leave out the type?

Reply:
Yes, the analyzer will complain, specifically about the following lint rule:

**&nbsp;&nbsp;lint • Type annotate public APIs at ... • type_annotate_public_apis**

With the increased importance of static type annotations in Dart 2, I feel that we get more benefits from enforcing the `type_annotate_public_apis` lint than any gains we achieve (like more terse code) from excluding it.

Without the return type, I find it easy to quickly scan the method signature `main() async { ... }` and "read" `main() { ... }`. The return type sticks out like a warning sign, alerting the reader to the fact that main is asynchronous. 

> It obscures the entry point

I find that the code highlighting helps make it just as easy to locate the entry point, because `main()` is in black font, whereas the tokens around it are more lightly colored.

Reply:
The difference between the type and the method name isn't very noticeable to me:

![image](https://user-images.githubusercontent.com/2164483/33858332-05334e0a-de84-11e7-94f3-d88f379993aa.png)

Reply:
@filiph what do you think we should do? I don't like turning off lints. Maybe we can get a special case for "main" in the lint?

Reply:
Just noting that this issue isn't gating. (We are still discussing this offline.)